.github/workflows/ci.yml

```diff
@@ -0,0 +1,34 @@
+name: ci
+on: [push, pull_request]
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.7', '3.8', '3.9', '3.10']
+    steps:
+      - uses: actions/checkout@v3
+      - run: pipx install poetry
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: 'poetry'
+      - run: poetry install -E with_everything
+      # - run: poetry show --all
+      # - run: poetry show --outdated
+      # - run: poetry show --tree
+      - run: poetry run pytest --benchmark-disable --cov --cov-report= tests/
+      - name: Check that Prospector can run
+        run: |
+          echo "Checking prospector can run; don't care about messages found, only that it runs successfully"
+          poetry run prospector --quiet --zero-exit
+          echo "Prospector ran successfully."
+      - name: Build and validate packages
```

Comment:
Maybe we could separate this part as it's "packaging check" not tests ?

Reply:
Line 22 is `Check that Prospector can run`
So should line 27 be `Check packaging` ?

Reply:
I think that the modern approach is just `poetry publish` instead of using `twine` but I do not know the pro & cons.
* https://python-poetry.org/docs/cli/#publish

Reply:
> I think that the modern approach is just poetry publish instead of using twine

The cons is that I don't know how poetry works and twine is "working ™️" :) The actual release is done in release.yml on tag published though, this is just a check here.